tests/v2/validation/provisioning/proxy/rke2_proxy_custom_cluster_test.go

```diff
@@ -0,0 +1,112 @@
+)
+
+type ProxyRKE2CustomClusterTestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	session           *session.Session
+	corralPackage     *corral.Packages
+	clustersConfig    *provisioninginput.Config
+	EnvVar            rkev1.EnvVar
+	corralImage       string
+	corralAutoCleanup bool
+}
+
+func (r *ProxyRKE2CustomClusterTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *ProxyRKE2CustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	r.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, r.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	r.clustersConfig.RKE2KubernetesVersions, err = kubernetesversions.Default(
+		r.client, clusters.RKE2ClusterType.String(), r.clustersConfig.RKE2KubernetesVersions)
+	require.NoError(r.T(), err)
+
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(r.T(), err)
+
+	corralConfig := corral.Configurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(r.T(), err)
+
+	r.corralPackage = corral.PackagesConfig()
+	r.corralImage = r.corralPackage.CorralPackageImages[corralPackageAirgapCustomClusterName]
+	r.corralAutoCleanup = r.corralPackage.HasCleanup
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryPrivateIP)
+		require.NoError(r.T(), err)
+
+		r.EnvVar.Name = "HTTP_PROXY"
+		r.EnvVar.Value = bastionIP + ":3219"
+		r.clustersConfig.AgentEnvVars = append(r.clustersConfig.AgentEnvVars, r.EnvVar)
+
+		r.EnvVar.Name = "HTTPS_PROXY"
+		r.EnvVar.Value = bastionIP + ":3219"
+		r.clustersConfig.AgentEnvVars = append(r.clustersConfig.AgentEnvVars, r.EnvVar)
+
+		r.EnvVar.Name = "NO_PROXY"
+		r.EnvVar.Value = "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc"
+		r.clustersConfig.AgentEnvVars = append(r.clustersConfig.AgentEnvVars, r.EnvVar)
+
+		err = corral.SetCorralSSHKeys(corralRancherHA.Name)
+		require.NoError(r.T(), err)
+
+		err = corral.SetCorralBastion(corralRancherHA.Name)
+		require.NoError(r.T(), err)
+	} else {
+		r.T().Logf("Using AgentEnvVars from config: %v", r.clustersConfig.AgentEnvVars)
+	}
+
+}
+
+func (r *ProxyRKE2CustomClusterTestSuite) TestProxyRKE2CustomClusterProvisioning() {
+	r.clustersConfig.MachinePools = []provisioninginput.MachinePools{provisioninginput.AllRolesMachinePool}
+
+	tests := []struct {
+		name    string
+		client  *rancher.Client
+		runFlag bool
+	}{
+		{provisioninginput.AdminClientName.String() + "-" + permutations.RKE2AirgapCluster + "-", r.client, r.client.Flags.GetValue(environmentflag.Short)},
```

Comment:
Same note with the admin user being standard instead.

Reply:
Updated for standard user